general/management/commands/gm_worker_async_tasks.py

```diff
@@ -95,7 +96,11 @@ def task_email_random_sound(self, gearman_worker, gearman_job):
         random_sound_id = gearman_job.data
         random_sound = Sound.objects.get(id=random_sound_id)
 
-        send_mail_template(\
+        disabled = UserEmailSetting.objects.filter(user=random_sound.user,
+                                email_type__name="random_sound").count()
```

Comment:
Even better, the functionality will always be the same:
if not disabled, send email.
So this should really be one function call:

if email_not_disabled(user, "something"):  # make this throw errors if "something" is an unknown email type!
    send_mail_template(...)